apps/rehabstod/src/pages/Patient/components/SelectRekoStatus/SelectRekoStatus.tsx

```diff
@@ -1,42 +1,44 @@
-import { useState } from 'react'
-import { useDispatch } from 'react-redux'
+import { useEffect, useState } from 'react'
 import { SelectFilter } from '../../../../components/Table/filter/SelectFilter'
-import { RekoStatus, RekoStatusType } from '../../../../schemas/sickLeaveSchema'
+import { RekoStatusType } from '../../../../schemas/sickLeaveSchema'
 import { useAppSelector } from '../../../../store/hooks'
-import { useSetRekoStatusMutation } from '../../../../store/sickLeaveApi'
+import { useGetRekoStatusForPatientQuery, useSetRekoStatusMutation } from '../../../../store/sickLeaveApi'
 import { getRekoStatusSickLeaveTimestamp } from '../../../../utils/getRekoStatusSickLeaveTimestamp'
-import { updateRekoStatusId } from '../../../../store/slices/sickLeave.slice'
 
 export function SelectRekoStatus({
   patientId,
   endDate,
+  startDate,
   rekoStatusTypes,
   description,
   disabled = false,
 }: {
   patientId: string
   endDate: string
+  startDate: string
   rekoStatusTypes: RekoStatusType[]
   description: string
   disabled?: boolean
 }) {
   const [setRekoStatus] = useSetRekoStatusMutation()
-  const { rekoStatusId } = useAppSelector((rootState) => rootState.sickLeave)
+  const { data: rekoStatusFromDatabase } = useGetRekoStatusForPatientQuery({ patientId, endDate, startDate })
```

Comment:
"FromDatabase" is a bit verbose, I guess everything from the API is stored in a database.